dubbo-common/src/main/java/org/apache/dubbo/common/utils/ClassHelper.java

```diff
@@ -265,26 +265,29 @@ public static boolean isPrimitive(Class<?> type) {
     }
 
     public static Object convertPrimitive(Class<?> type, String value) {
-        if (type == char.class || type == Character.class) {
+        if (value == null) {
+            return null;
+        } else if (type == char.class || type == Character.class) {
             return value.length() > 0 ? value.charAt(0) : '\0';
         } else if (type == boolean.class || type == Boolean.class) {
             return Boolean.valueOf(value);
         } else if (type == byte.class || type == Byte.class) {
-            return Byte.valueOf(value);
+            return StringUtils.isNumeric(value, false) ? Byte.valueOf(value) : null;
```

Comment:
I am not persuaded that 'StringUtils.isNumeric()' is required here. In my opinion, JDK has already offered the nice API and we should stick to it. What we need to do is to catch **NumberFormatException** and return 'null' if necessary.

Reply:
I think this is good, but I think the function enhancement of `StringUtils.isNumeric()` is not necessary to delete it. What do you think?